syft/core/utils.py

```diff
@@ -1,5 +1,377 @@
+                                                             skip_register=True)
+                                if syft_obj.torch_type == 'syft.Variable' and 'child' is not None:
+                                    syft_obj.parent.data.parent = obj['child']
+
+                            else:
+                                # We recreate the pointer to be transmitted
+                                owner = self.worker.get_worker(obj['owner'])
+                                syft_obj = sy._PointerTensor(child=None,
+                                                             parent=None,
+                                                             torch_type=obj['torch_type'],
+                                                             location=obj['location'],
+                                                             id_at_location=obj['id_at_location'],
+                                                             owner=owner,
+                                                             id=obj['id'],
+                                                             skip_register=True)
+                            return syft_obj
+                    else:
+                        raise Exception('SyftTensor', obj_type, 'is not supported so far')
+                # Case of a iter type non json serializable
+                elif obj_type in ('tuple', 'set', 'bytearray', 'range'):
+                    return eval(obj_type)(obj)
+                # Case of a slice
+                elif obj_type == 'slice':
+                    return slice(*obj['args'])
+                # Case of a worker
+                elif obj_type == 'worker':
+                    return self.worker.get_worker(obj)
+                else:
+                    return obj
+            else:
+                pass
+        return dct
+
+
+def compile_command(attr, args, kwargs, has_self=False, self=None):
+    command = {
+        'command': attr,
+        'has_self': has_self,
+        'args': args,
+        'kwargs': kwargs
+    }
+    if has_self:
+        command['self'] = self
+
+    encoder = PythonEncoder()
+    command, tensorvars, pointers = encoder.encode(command, retrieve_tensorvar=True, retrieve_pointers=True)
+
+    # Get information about the location and owner of the pointers
+    locations = []
+    owners = []
+    for pointer in pointers:
+        locations.append(pointer.location)
+        owners.append(pointer.owner)
+    locations = list(set(locations))
+    owners = list(set(owners))
+
+    if len(locations) > 1:
+        raise Exception('All pointers should point to the same worker')
+    if len(owners) > 1:
+        raise Exception('All pointers should share the same owner.')
+
+    return command, locations, owners
+
+
+def convert_local_syft_to_torch(obj, reverse=False, worker=None):
+    # Case of basic types and slice
+    if isinstance(obj, (int, float, str, slice)) or obj is None:
```

Comment:
This design pattern makes this function very hard to test/ understand what it is doing. Is there a way that we can modify this to avoid all of the different if/else cases